File: tools/memory/retrieveMemory.js
/**
 * Tool: Retrieve Memory
 * Retrieves information from short-term memory
 */
export default {
  name: 'retrieveMemory',
  description: 'Retrieve information from short-term memory',
  parameters: { key: 'string' },
  
  async execute(params, context) {
    const { key } = params;
    const { memory } = context;
    
    const entry = memory.entries.find(e => e.key === key);
    
    if (!entry) {
      return {
        success: false,
        message: `Key '${key}' not found in short-term memory`,
        value: null
      };
    }
    
    return {
      success: true,
      key: entry.key,
      value: entry.value,
      metadata: entry.metadata
    };
  }
};

File: obscurex/tools/memory/retrieve_memory.py
"""
Tool: Retrieve Memory
Retrieves information from short-term memory
"""

from typing import Dict, Any


async def execute(params: Dict[str, Any], context: Dict[str, Any]) -> Dict[str, Any]:
    """Retrieve information from short-term memory."""
    key = params.get('key')
    memory = context['memory']
    
    entry = None
    for e in memory['entries']:
        if e['key'] == key:
            entry = e
            break
    
    if not entry:
        return {
            'success': False,
            'message': f"Key '{key}' not found in short-term memory",
            'value': None
        }
    
    return {
        'success': True,
        'key': entry['key'],
        'value': entry['value'],
        'metadata': entry['metadata']
    }


# Tool definition
tool = {
    'name': 'retrieveMemory',
    'description': 'Retrieve information from short-term memory',
    'parameters': {'key': 'string'},
    'execute': execute
}
